src/main/java/org/truffleruby/core/RaiseIfFrozenNode.java

```diff
@@ -32,11 +33,19 @@ public RaiseIfFrozenNode(RubyNode child) {
     public Object execute(VirtualFrame frame) {
         Object result = child.execute(frame);
 
-        if (isFrozenNode.executeIsFrozen(result)) {
+        if (getIsFrozen().executeIsFrozen(result)) {
             errorProfile.enter();
             throw new RaiseException(coreExceptions().frozenError(result, this));
         }
 
         return result;
     }
+
+    private IsFrozenNode getIsFrozen() {
```

Comment:
This should be `isFrozen` and include the call to `executeIsFrozen(result)`, so it's nicer at the call site.
(From https://github.com/oracle/truffleruby/blob/master/doc/contributor/code-patterns.md#the-lazy-pattern)

Reply:
thanks, fixed.